### DISCUSSION TITLE
BCBS MRF Files

### DISCUSSION THREAD

**Discussion by TiaGossSawhney** [CMSgov/price-transparency-guide#616] [General]
**Some BCBS reporting entities have more than 1,000 files.  Why? And what do you need to watch out for?  How do you find the files of most interest?**

To understand the files, you need to understand how the BCBS organization work.  Each local organization, for example Horizon BCBS of New Jersey, establishes it's own networks and it's own negotiated prices.  They offer 'local plans' where in-network is composed of providers in one of their own networks.  Horizon's networks are almost entirely New Jersey providers, with a few Philadelphia and New York City provider exceptions.  (Horizon, like most carriers, has wider and narrower networks.)

But national employers and even some smaller groups and individuals want a national plan and wide access to providers.  Enter "Blue Card" coverage.  With Blue Card coverage, the insured has  in-network access to a wide network from every BCBS organization.  

Transparency then means providing MRF files from every BCBS organization in the country, which creates a jumbled mess:

1)  Every BCBS organization independently creates their files, which means that they are inconsistent.  Some may not validate.  Some may be missing data.  Etc.

2)  There may be non-BCBS organizations in the mix, with their own reporting inconsistencies, such as carve out mental health organizations.

3)  The BCBS organizations generally each have multiple MRF files.

If you are lucky, you might be most interested in the MRFs for only certain states.  There are two ways to find a MRF for a particular state.  

1)  BCBS "plan number" is embedded within most of the file.  It's the three digits after the date.  For example, it is the "890" (Tennessee) in https://bcbsmn.mrf.bcbs.com/2022-10_890_58B0_in-network-rates_64_of_64.json.gz (a BCBS Minnesota MRF).  This unfortunately old table allows you to look up most of the plan numbers:  https://www.health.ny.gov/statistics/sparcs/sysdoc/appl.htm.  (Let me know if you have a newer table!)

2) You can scan each of the tables.  _Reporting_entity_name_ is the first attribute of each table.  

Finally, Blue Card MRFs should be the same irrespective of which BCBS organization is providing the files.  Local plans (non-Blue Card), however, will be different for each BCBS organization.

**Comment by js1975** [comment]
The example filename ends with _in-network-rates_64_of_64.json.gz  
In the table of contents file we other MRF with _01_of_64 , _02_of_64  through _64_of_64 .

What is the underlying logic by which payers segment their data when they create a series of MRF like this?
Is the pattern the same across payers?

**Reply by hw-monkey** [reply]
There is a pattern. It is by plan. And _sometimes_ the file name must follow a naming convention. Why in the world have a specification that is only sometimes? I do not know. See below...
____________________________________________________________
Only alphanumeric characters are allowed in the file name. No special characters such as ' are allowed. Special characters are either to be removed completely or replaced with -.

Single Plan Files
**The following is the required naming standard for each file: `<YYYY-MM-DD>_<payer or issuer name>_<plan name>_<file type name>.<file extension>`**

For example, the following would be the required naming for CMS building a JSON file:

2020-01-05_cms_medicare_in-network-rates.json
2020-01-05_cms_medicare_allowed-amounts.json
An example of a plan named healthcare 100 with an issuer's name issuer abc producing a JSON file, the following would be the naming output:

2020-01-05_issuer-abc_healthcare-100_in-network-rates.json
2020-01-05_issuer-abc_healthcare-100_allowed-amounts.json

Multiple Plans Per File
If multiple plans are to be included in a single file, a table-of-contents file will be required. _The naming standard will be applied to the table-of-contents file_ and **both the in-network and allowed-amounts files will not have any naming standards.**

The following is the required naming standard for the table-of-contents file: `<YYYY-MM-DD>_<payer or issuer name>_index.<file extension>`

For example, the following would be the required naming for CMS building a JSON file that includes Medicare and Medicaid plans:

2020-01-05_cms_index.json
